### DISCUSSION TITLE
Scripts with user settings

### DISCUSSION THREAD

**Discussion by Z4rko963** [chhoumann/quickadd#657] [Q&A] [answered]
Hello, I hope someone can assist me. I've been trying for a while to accomplish this task without success. The values are being prompted one by one, and I can't manage to display all of them together at once, only sequentially. Thanks.

![script_with_settings-0141e6de7c4ab40e8697d99f9ebeeb7b](https://github.com/chhoumann/quickadd/assets/146877893/5b30bbb0-3d7a-4a72-8fc1-64a10393b9a8)

```js quickadd
const myScript = {
  settings: {
    text: "",
    checkbox: false,
    dropdown: "Option 1",
    format: "{{DATE:YYYY-MM-DD}}"
  },

  myQuickAddFunction: async function() {
    // Aquí puedes acceder a la API de QuickAdd y usar las configuraciones definidas
    const textInput = await this.api.inputPrompt("Enter some text");
    const checkboxResult = await this.api.yesNoPrompt("Confirm?");
    const dropdownChoice = await this.api.suggester(["Option 1", "Option 2", "Option 3"], ["Option 1", "Option 2", "Option 3"]);
    const currentDate = await this.api.date.now("YYYY-MM-DD");

    // Procesa los resultados como necesites
    console.log(textInput, checkboxResult, dropdownChoice, currentDate);
  }
};

// Suponiendo que 'api' es la API de QuickAdd ya disponible en el contexto
myScript.api = app.plugins.plugins.quickadd.api;

// Ejecutar la función dentro del script
myScript.myQuickAddFunction();
```

**Comment by chhoumann** [comment, marked as answer]
Hey @Z4rko963 

Normal QuickAdd behavior is to ask sequentially when just using the plugin. However, I added the possibility of creating script settings which would show all the settings at once - as expected.

You can find a script that shows script settings here:
https://quickadd.obsidian.guide/docs/Advanced/scriptsWithSettings

Here's a video to clarify the behavior:

https://github.com/chhoumann/quickadd/assets/29108628/5abdb72d-0b64-456a-b066-da3596084a2b

**Comment by Z4rko963** [comment]
Hi @chhoumann ,

I just wanted to express my gratitude for the clarification on QuickAdd's behavior. It's clear now that the default is to prompt sequentially, and the option to script settings for an all-at-once display is a fantastic addition. The script settings link and the video were particularly helpful. Thanks for making QuickAdd such a powerful tool!. Thanks.
